### PR TITLE
Add `SkipRoslynAnalyzers` parameter to `Start-PSBuild`

### LINKED ISSUE #15632: Add `-SkipAnalyzers` switch parameter to `Start-PSBuild`
> @iSazonov Perhaps we could add `-SkipAnalyzers` switch parameter to `Start-PSBuild`?

Good point.

_Originally posted by @iSazonov in https://github.com/PowerShell/PowerShell/issues/14336#issuecomment-749320265_

### PR DESCRIPTION
**Pull request by xtqqczze**
Fix #15632

### CODE REVIEW

**Comment by iSazonov** [comment]
It is not clear will this block source generators?

**Comment by xtqqczze** [comment]
> It is not clear will this block source generators?

I haven't found any evidence it will.

File: build.psm1 [hunk 1]
```diff
@@ -307,7 +307,8 @@ function Start-PSBuild {
         [ValidateNotNullOrEmpty()]
         [string]$ReleaseTag,
         [switch]$Detailed,
-        [switch]$InteractiveAuth
+        [switch]$InteractiveAuth,
+        [switch]$SkipAnalyzers
```
Comment (iSazonov) [inline build.psm1:311]: Maybe:
```suggestion
        [switch]$SkipCSharpAnalyzers
```
I think about PSSA :-)

Comment (xtqqczze) [reply build.psm1:311]: Not all analyzers are C#, e.g.: [IL3000](https://docs.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/il3000).

I prefer `SkipRoslynAnalyzers` in this case.

**Review by iSazonov — APPROVED** [review]

File: build.psm1 [hunk 2]
```diff
@@ -453,6 +454,13 @@ Fix steps:
         $Arguments += "/property:ReleaseTag=$ReleaseTagToUse"
     }
 
+    if (-not $SkipRoslynAnalyzers) {
+        $Arguments += "/property:RunAnalyzersDuringBuild=true"
+    }
+    else {
+        $Arguments += "/property:RunAnalyzersDuringBuild=false"
+    }
+
```
Comment (iSazonov) [inline build.psm1:463]:
```suggestion
    if ($SkipRoslynAnalyzers) {
        $Arguments += "/property:RunAnalyzersDuringBuild=false"
    }
    else {
        $Arguments += "/property:RunAnalyzersDuringBuild=true"
    }

```

Comment (xtqqczze) [reply build.psm1]: I removed `RunAnalyzersDuringBuild=true` altogether as this could override setting in csproj.

**Comment by iSazonov** [comment]
@xtqqczze Thanks!

**Pull request merged**